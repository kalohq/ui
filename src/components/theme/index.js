/* @flow */
export {default} from './theme';
