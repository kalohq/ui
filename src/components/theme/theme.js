/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
