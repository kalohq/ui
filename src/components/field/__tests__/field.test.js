/* eslint-env jest */
import {shallow} from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import Field from '../field.react';
import Input from '../../input';
import theme from '../../theme';

describe('Field', () => {
  const defaultProps = {
    label: 'My Label',
    hint: 'My Hint',
  };

  const create = props => (
    <Field {...defaultProps} {...props}>
      <ThemeProvider theme={theme}>
        <Input value="My Input Value" />
      </ThemeProvider>
    </Field>
  );

  test('should render shallow component ok', () => {
    const element = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Field {...defaultProps}>
            <Input value="My Input Value" />
          </Field>
        </ThemeProvider>
      )
      .toJSON();
    expect(element).toMatchSnapshot();
  });

  test('should render as expected', () => {
    const element = create();
    const result = shallow(element);
    expect(result.html()).toMatch(/My Label/);
    expect(result.html()).toMatch(/My Hint/);
    expect(result.find(Input).is({value: 'My Input Value'})).toBe(true);
  });

  test('should render with validations', () => {
    const element = create({
      validations: [
        {
          message: 'Your email was incorrect',
        },
      ],
    });
    const result = shallow(element);
    expect(result.html()).toMatch(/Your email was incorrect/);
  });
});
