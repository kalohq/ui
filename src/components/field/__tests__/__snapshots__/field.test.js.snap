// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field should render shallow component ok 1`] = `
<div
  className="ui-field ui-field--legacy e1krcyn16 css-0"
  onClick={undefined}
  style={Object {}}
>
  <div
    className="e1krcyn10 css-1dvaom5"
    style={Object {}}
  >
    <label
      className="ui-field-label e1krcyn16 css-4or71o"
      htmlFor={undefined}
      style={Object {}}
      title={undefined}
    >
      My Label
    </label>
  </div>
  <div
    className="ui-input-container ui-input-container--margin-none"
  >
    <input
      className="ui-input ui-input--medium"
      defaultValue={undefined}
      disabled={false}
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      readOnly={false}
      style={
        Object {
          "width": undefined,
        }
      }
      value="My Input Value"
    />
  </div>
  <span
    className="ui-field-hint e1krcyn16 css-0"
    style={Object {}}
  >
    My Hint
  </span>
</div>
`;
